src/tests/ftest/osa/osa_online_extend.py

```diff
@@ -0,0 +1,248 @@
+        rank = total_servers
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context,
+                                 dmg_command=self.get_dmg_command())
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+
+        # Extend the pool_uuid, rank and targets
+        for val in range(0, num_pool):
+            for oclass, api, test, flags in product(self.ior_daos_oclass,
+                                                    self.ior_apis,
+                                                    self.ior_test_sequence,
+                                                    self.ior_flags):
+                threads = []
+                for thrd in range(0, num_jobs):
+                    # Add a thread for these IOR arguments
+                    threads.append(threading.Thread(target=self.ior_thread,
+                                                    kwargs={"pool": pool[val],
+                                                            "oclass": oclass,
+                                                            "api": api,
+                                                            "test": test,
+                                                            "flags": flags,
+                                                            "results":
+                                                            self.out_queue}))
+                # Launch the IOR threads
+                for thrd in threads:
+                    self.log.info("Thread : %s", thrd)
+                    thrd.start()
+                    time.sleep(5)
+            self.pool = pool[val]
+            scm_size = self.pool.scm_size
+            nvme_size = self.pool.nvme_size
+            self.pool.display_pool_daos_space("Pool space: Beginning")
+            pver_begin = self.get_pool_version()
+
+            # Start the additional servers and extend the pool
+            self.log.info("Extra Servers = %s", self.extra_servers)
+            self.start_additional_servers(self.extra_servers)
+            # Give sometime for the additional server to come up.
+            time.sleep(5)
+            self.log.info("Pool Version at the beginning %s", pver_begin)
+            output = self.dmg_command.pool_extend(self.pool.uuid,
+                                                  rank, scm_size,
+                                                  nvme_size)
+            self.log.info(output)
+
+            fail_count = 0
+            while fail_count <= 20:
+                pver_extend = self.get_pool_version()
+                rebuild_status = self.get_rebuild_status()
```

Comment:
(pylint-no-member) Instance of 'OSAOnlineExtend' has no 'get_rebuild_status' member
(pylint-unused-variable) Unused variable 'rebuild_status'